system/modules/task/actions/edit.php

```diff
@@ -83,13 +83,14 @@ function edit_GET($w) {
             ),
 			array(
 				array("Estimated hours", "text", "estimate_hours", $task->estimate_hours),
-				array("Effort", "text", "effort", $task->effort)
+				array("Effort", "text", "effort", $task->effort),
+                            ((!empty($p["id"]) && $task->canISetRate()) ? (new InputField())->setName('rate')->setLabel('Rate ($)')->setValue($task->rate)->setPattern('^\d+(?:\.\d{1,2})?$')->setPlaceholder('0.00') : null)
```

Comment:
I know the code around this checks for an empty $p['id'] but thats actually not the right way to do it. We should be checking for an empty $task->id. When you give the action a task_id that doesn't exist in the data base, in this code, $p['id'] will have something in it, but the task will be null, so all calls to it will still break. This is only ever an issue if someone manually changes the task id in the url, or if tasks are linked to incorrectly. So not a huge issue but probably one that we should look at soon